src/types.ts

```diff
@@ -5,6 +5,7 @@ export interface ProducerOptions {
   batchSize?: number;
   sqs?: SQSClient;
   region?: string;
+  useQueueUrlAsEndpoint?: boolean;
```

Comment:
I wonder if it's better just to extend SQS' native types here rather than implementing a new type every time they have a new option.

Reply:
It can be extended if it's ok to update @aws-sdk/client-sqs version to at least 3.507 where this functionality is added.

Reply:
In this way handler can be passed, that is not what we what from producer

Reply:
Sure, we can look at this comment separately, just a comment on future prosperity.

Reply:
Note to self: After merging I think that our types need some comments and documentation like we have for sqs-consumer.